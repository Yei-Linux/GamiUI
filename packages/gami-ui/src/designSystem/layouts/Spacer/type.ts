import { TDesignUI } from 'core/domain/types/design-system'

export type TDirection = 'left' | 'right' | 'top' | 'bottom'

export type TSize = 1 | 2 | 3 | 4 | 5

export type TSpacer = {
  /**
   * Spacer Direction
   */
  direction: TDirection
  /**
   * Size of spacer
   */
  size?: TSize
  /**
   * Custom Size
   */
  customSize?: string
} & TDesignUI
