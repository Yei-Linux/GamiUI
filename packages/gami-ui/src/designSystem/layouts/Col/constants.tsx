import { ColSpacingTypes } from 'core/domain/types'

export const options = {
  spacingType: ['none', 'sm', 'md', 'lg', 'custom'] as const,
  type: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12] as const,
}

export const spacingValues: Record<ColSpacingTypes, string> = {
  sm: '8px',
  md: '16px',
  lg: '24px',
  none: '0px',
  custom: 'custom',
}
