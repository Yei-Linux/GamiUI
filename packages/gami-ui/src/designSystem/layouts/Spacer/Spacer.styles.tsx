import { css } from '@emotion/react'
import styled from '@emotion/styled'
import { TSpacer } from './type'

const directionSpaceCss = ({
  left = '0px',
  right = '0px',
  top = '0px',
  bottom = '0px',
}) => css`
  margin: ${`${top} ${right} ${bottom} ${left}`};
`

type TSpacerStyled = {
  $customSize: TSpacer['customSize']
  $direction: TSpacer['direction']
  $size: TSpacer['size']
}
export const SpacerStyled = styled('div')(
  ({ $customSize, $direction, $size }: TSpacerStyled) =>
    directionSpaceCss({
      [$direction]: `${$customSize || `${$size}rem`}`,
    })
)
