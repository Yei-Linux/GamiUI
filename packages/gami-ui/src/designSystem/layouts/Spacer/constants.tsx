export const options = {
  directionType: ['left', 'right', 'top', 'bottom'] as const,
  sizeType: [1, 2, 3, 4, 5] as const,
  type: [] as const,
}
