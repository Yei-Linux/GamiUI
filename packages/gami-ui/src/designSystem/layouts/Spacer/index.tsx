export { default } from './Spacer'
export { options } from './constants'
