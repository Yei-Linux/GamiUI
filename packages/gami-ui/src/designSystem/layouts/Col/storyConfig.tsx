import {
  getInheritGlobalStylesStories,
  getStoryConfigStructure,
} from 'core/helpers/storybook.helper'
import Col, { options } from '.'
import {
  IStoryConfig,
  IStoryMainConfig,
} from 'core/domain/interfaces/IStorybook'

const storyArgTypes = {
  className: {
    control: 'text',
    description: 'Classname Prop',
    table: {
      type: { summary: 'string' },
      defaultValue: Col.defaultProps?.className,
    },
  },
  spacing: {
    control: { type: 'select', options: options.spacingType },
    description: 'Spacing Prop',
    table: {
      type: { summary: 'string' },
      defaultValue: Col.defaultProps?.spacing,
    },
  },
  customSpacing: {
    control: 'text',
    description: 'Custom Spacing Prop',
    table: {
      type: { summary: 'string' },
      defaultValue: Col.defaultProps?.customSpacing,
    },
  },
  xs: {
    control: { type: 'select', options: options.type },
    description: 'Extra Small Size',
    table: {
      type: { summary: 'string' },
      defaultValue: Col.defaultProps?.xs,
    },
  },
  sm: {
    control: { type: 'select', options: options.type },
    description: 'Small Size',
    table: {
      type: { summary: 'string' },
      defaultValue: Col.defaultProps?.sm,
    },
  },
  md: {
    control: { type: 'select', options: options.type },
    description: 'Medium Size',
    table: {
      type: { summary: 'string' },
      defaultValue: Col.defaultProps?.md,
    },
  },
  lg: {
    control: { type: 'select', options: options.type },
    description: 'Large Size',
    table: {
      type: { summary: 'string' },
      defaultValue: Col.defaultProps?.lg,
    },
  },
}

const docArgTypes = {
  children: {
    control: 'object',
    description: 'Children Prop',
    table: {
      type: { summary: 'React.ReactNode' },
    },
  },
}

const argTypes = { ...storyArgTypes, ...docArgTypes }

const mainConfig: IStoryMainConfig = {
  title: 'Layout/Col ðŸŸ¢',
  component: Col,
  args: {},
  argTypes: storyArgTypes,
  parameters: {
    actions: { argTypesRegex: '^on[A-Z].*' },
    docs: {
      source: {
        type: 'code',
      },
    },
  },
}

const storiesInheritGlobalStyles = getInheritGlobalStylesStories({
  rounded: null,
  shadow: null,
  width: null,
  height: null,
  size: null,
  fontWeight: null,
  textAlign: null,
  margin: null,
  padding: null,
})

const storiesComponent: IStoryConfig[] = []

const storyConfig = getStoryConfigStructure({
  mainConfig,
  storiesInheritGlobalStyles,
  storiesComponent,
  component: Col,
})

export { storyConfig, argTypes }
