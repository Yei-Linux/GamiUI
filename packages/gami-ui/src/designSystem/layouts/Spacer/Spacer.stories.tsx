import React from 'react'
import { Meta, ComponentStory } from '@storybook/react'
import { TJSXElements } from 'core/domain/interfaces/common'

import { storyConfig } from './storyConfig'
import Spacer from '.'
import Row from 'designSystem/layouts/Row'
import Avatar from 'designSystem/atoms/Avatar'
import { avatarProfilesImages } from 'designSystem/atoms/Avatar/constants'

const { mainConfig, stories } = storyConfig

export default mainConfig as Meta

// eslint-disable-next-line no-empty-pattern
const [] = stories

const StorySizeDirection = (args: any) => (
  <Row gap="10px">
    <Avatar src={avatarProfilesImages.one} rounded="lg" />
    <Spacer direction="left" size={1} {...args} />
    <Avatar src={avatarProfilesImages.one} rounded="lg" />
  </Row>
)

export const StorySizeDirectionTemplate: ComponentStory<TJSXElements> =
  StorySizeDirection.bind({})
StorySizeDirectionTemplate.storyName = 'WithSizeDirection'
StorySizeDirectionTemplate.args = {
  direction: 'left',
  size: 1,
}

const StoryCustomSizeDirection = (args: any) => (
  <Row gap="10px">
    <Avatar src={avatarProfilesImages.one} rounded="lg" />
    <Spacer direction="left" customSize="45px" {...args} />
    <Avatar src={avatarProfilesImages.one} rounded="lg" />
  </Row>
)

export const StoryCustomSizeDirectionTemplate: ComponentStory<TJSXElements> =
  StoryCustomSizeDirection.bind({})
StoryCustomSizeDirectionTemplate.storyName = 'WithCustomSizeDirection'
StoryCustomSizeDirectionTemplate.args = {
  direction: 'left',
  customSize: '45px',
}
