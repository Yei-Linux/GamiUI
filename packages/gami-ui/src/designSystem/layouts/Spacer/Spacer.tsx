import React from 'react'
import * as S from './Spacer.styles'

export type TDirection = 'left' | 'right' | 'top' | 'bottom'

export type TSize = 1 | 2 | 3 | 4 | 5

export interface ISpacer {
  /**
   * Spacer Direction
   */
  direction: TDirection
  /**
   * Size of spacer
   */
  size?: TSize
  /**
   * Custom Size
   */
  customSize?: string
  /**
   * Classname
   */
  className?: string
}

const Spacer = ({ direction, size = 1, customSize, className }: ISpacer) => {
  return (
    <S.Spacer
      $direction={direction}
      $size={size}
      $customSize={customSize}
      className={className}
    />
  )
}

Spacer.displayName = 'Spacer'

export default Spacer
