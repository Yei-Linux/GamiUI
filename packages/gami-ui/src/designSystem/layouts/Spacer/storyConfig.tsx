import {
  getInheritGlobalStylesStories,
  getStoryConfigStructure,
} from 'core/helpers/storybook.helper'
import Spacer, { options } from '.'
import {
  IStoryConfig,
  IStoryMainConfig,
} from 'core/domain/interfaces/IStorybook'

const storyArgTypes = {
  className: {
    control: 'text',
    description: 'Classname Prop',
    table: {
      type: { summary: 'string' },
    },
  },
  direction: {
    control: { type: 'select', options: options.directionType },
    description: 'Spacer Direction',
    table: {
      type: { summary: 'string' },
    },
  },
  size: {
    control: { type: 'select', options: options.sizeType },
    description: 'Size of spacer',
    table: {
      type: { summary: 'string' },
    },
  },
  customSize: {
    control: 'text',
    description: 'Custom Size',
    table: {
      type: { summary: 'string' },
    },
  },
}

const docArgTypes = {
  children: {
    control: 'object',
    description: 'Children Prop',
    table: {
      type: { summary: 'React.ReactNode' },
    },
  },
}

const argTypes = { ...storyArgTypes, ...docArgTypes }

const mainConfig: IStoryMainConfig = {
  title: 'Layout/Spacer ðŸŸ¢',
  component: Spacer,
  args: {},
  argTypes: storyArgTypes,
}

const storiesInheritGlobalStyles = getInheritGlobalStylesStories({
  rounded: null,
  shadow: null,
  width: null,
  height: null,
  size: null,
  fontWeight: null,
  textAlign: null,
  margin: null,
  padding: null,
})

const storiesComponent: IStoryConfig[] = []

const storyConfig = getStoryConfigStructure({
  mainConfig,
  storiesInheritGlobalStyles,
  storiesComponent,
  component: Spacer,
})

export { storyConfig, argTypes }
