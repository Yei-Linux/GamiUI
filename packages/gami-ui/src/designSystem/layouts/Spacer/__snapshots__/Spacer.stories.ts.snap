// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Layout/Spacer Basic 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
    }
  }
>
  <div
    className="css-1e6xljk-RowWrapper e1y7zrl70"
    height="100%"
    style={Object {}}
  >
    <div
      className="css-i0h58x-ColWrapper e17m19uc0"
      spacing="0px"
    >
      <div
        className="css-psf7un-AvatarWrapper eklgmvl0"
        fontWeight="BOLD"
      >
        <img
          alt="Avatar"
          className="css-1moa472-Img efl6h300"
          height="40px"
          src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRPegjNNBzO3-k5t6hTm1htv1kxyWSOtCRr3A&usqp=CAU"
          width="40px"
        />
      </div>
    </div>
    <div
      className="css-1c3s0wr-SpacerWrapper e1h4jglc0"
      direction="right"
      size={1}
    />
    <div
      className="css-i0h58x-ColWrapper e17m19uc0"
      spacing="0px"
    >
      <div
        className="css-psf7un-AvatarWrapper eklgmvl0"
        fontWeight="BOLD"
      >
        <img
          alt="Avatar"
          className="css-1moa472-Img efl6h300"
          height="40px"
          src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRPegjNNBzO3-k5t6hTm1htv1kxyWSOtCRr3A&usqp=CAU"
          width="40px"
        />
      </div>
    </div>
  </div>
</div>
`;
